File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>((a, b) -> b - a);

        for(int num : nums) {
            pq.offer(num);
        }

        long score = 0;

        while(!pq.isEmpty() && k > 0) {
            int maxi = pq.poll();
            score += maxi;

            pq.add((int) (maxi + 2) / 3);
            k--;
        }

        return score;
    }
}

File: 2616-maximal-score-after-applying-k-operations/2616-maximal-score-after-applying-k-operations.py
class Solution:
    def maxKelements(self, nums: List[int], k: int) -> int:
        # O(n + klogn)
        score = 0
        max_heap = [-n for n in nums]
        heapq.heapify(max_heap)

        while k:
            maxi = abs(heapq.heappop(max_heap))
            score += maxi
            heapq.heappush(max_heap, -math.ceil(maxi / 3))
            k -= 1

        return score
        